e2e/cli-e2e/tests/history.e2e.test.ts

```diff
@@ -0,0 +1,53 @@
+import { readdir } from 'node:fs/promises';
+import { join } from 'node:path';
+import { PluginReport, Report, reportSchema } from '@code-pushup/models';
+import { cleanTestFolder } from '@code-pushup/test-setup';
+import { executeProcess, readJsonFile } from '@code-pushup/utils';
+
+describe('CLI history', () => {
+  /* eslint-disable @typescript-eslint/no-unused-vars */
+  const omitVariableData = ({
+    date,
+    duration,
+    version,
+    ...report
+  }: Omit<Report, 'commit'> | PluginReport) => report;
+  const omitVariableReportData = ({ commit, ...report }: Report) =>
+    omitVariableData({
+      ...report,
+      plugins: report.plugins.map(omitVariableData) as PluginReport[],
+    });
+  /* eslint-enable @typescript-eslint/no-unused-vars */
+
+  beforeEach(async () => {
+    await cleanTestFolder('tmp/e2e');
+  });
+
+  it('should run ESLint plugin and create report.json for the last 5 commits', async () => {
+    const { code, stderr } = await executeProcess({
+      command: 'code-pushup',
+      args: [
+        'history',
+        '--persist.outputDir=../../tmp/e2e/react-todos-app/history',
+        '--no-progress',
+        '--onlyPlugins=eslint',
+        '--forceCleanStatus=eslint',
+      ],
+      cwd: 'examples/react-todos-app',
+    });
+
+    expect(code).toBe(0);
+    expect(stderr).toBe('');
+
+    const outputDirFromRoot = join('tmp', 'e2e', 'react-todos-app', 'history');
+    const reportPaths = await readdir(outputDirFromRoot);
+    const results = await Promise.all(
+      reportPaths.map(path => readJsonFile(join(outputDirFromRoot, path))),
+    );
+
+    expect(results).toHaveLength(5);
+    results.forEach(report => {
+      expect(() => reportSchema.parse(report)).not.toThrow();
+    });
```

Comment:
There should be no loops in tests (same for `map` above). If only 2 commits are chosen, it is clear how many reports the commands run for. Now it wasn't even clear to me until I read the docs that 5 commits are the default number - 5 seemed like a random number.

Reply:
added 5 random `.at(POSITION)` to avoid the loop. same for the `map`.